providers/mysql_basic.rb

```diff
@@ -7,82 +7,46 @@
 
   ndb_mysql_basic "mysqld_start_grants" do
     wait_time 20
-    remove_mycnf 1
     action :wait_until_started
   end
 
-
   grants_path = "#{node['ndb']['base_dir']}/grants.sql"
+  template grants_path do
+    source "grants.erb"
+    owner node['ndb']['user']
+    mode "0600"
+    action :create_if_missing
+    variables({
+      :my_ip => my_ip
+    })
+  end
 
   exec= node['ndb']['scripts_dir'] + "/mysql-client.sh"
   bash 'run_grants' do
     user node['ndb']['user']
+    environment ({
+      'MYSQL_HOME' => node['ndb']['root_dir']
+    })
     code <<-EOF
-     set -e
-     export MYSQL_HOME=#{node['ndb']['root_dir']}
      #{exec} -e "source #{grants_path}"
     EOF
-    new_resource.updated_by_last_action(true)
     not_if "#{node['mysql']['base_dir']}/bin/mysql -u root --skip-password -S #{node['ndb']['mysql_socket']} -e \"SELECT user FROM mysql.user WHERE user=\"#{node['mysql']['user']}\"\"", :user => "#{node['ndb']['user']}"
   end
-
+  new_resource.updated_by_last_action(true)
 end
 
 action :wait_until_started do
 
-# mysql_install_db makes a copy of the my.cnf file in /etc/mysql. Remove it.
-# FC021 for the next line's new_resource.name attribute
-bash "remove_mycnf_#{new_resource.name}" do
-    user "root"
-    code <<-EOF
-      rm -f /etc/mysql/my.cnf  
-    EOF
-    only_if { new_resource.remove_mycnf == 1 }
```

Comment:
this was needed where people installed on a dirty machine with an existing mysql installation.  I don't see the need to remove it, it only does good stuff. MySQL will look first in /etc/mysql/my.cnf , no matter if you supply -f or not.

Reply:
Ok. I got confused by the comment. I thought that the `mysql_install_db` action we have later was creating a a copy of the my.cnf configuration as backup. I couldn't find it, so I assumed it has been removed and so I cleaned up also this part of the code.  - I can put it back